### ISSUE TITLE
Way to freeze a live table

### ISSUE THREAD

**Issue by devonpallison** [deephaven/deephaven-core#4829]
As a user, I want to be able to freeze a ticking table so that I can improve the performance of my query. We have a large table that is getting data in distinct "snapshots”. Querying arbitrary old snapshots can be quite slow even though the data for a given snapshot has been completely ingested. This is especially true when we end up taking the readLock over and over:

by(NORMAL:[ColumnAggregations{spec=AggSpecSum{}, pairs=[ColumnName(...)]) Concurrent snapshot function took 3743ms, snapshotSuccessful=false, functionSuccessful=true, beforeClockValue=1365, afterClockValue=1367, usePrev=false

![image](https://github.com/deephaven/deephaven-core/assets/62663619/52e37392-45f5-4343-83c9-31698390a29d)

It would be nice to have a way to treat these older snapshots as “dead” so that querying them is faster

**Comment by rcaudy** [comment]
This request doesn't seem particularly well-formed, from the context I have. What you really want is the ability to assert that a target region of a table is effectively static for purposes of snapshotting.

**Comment by devonpallison** [comment]
> This request doesn't seem particularly well-formed, from the context I have. What you really want is the ability to assert that a target region of a table is effectively static for purposes of snapshotting.

Yes, I think this would cover what I want. Though Charles added that, from a UI perspective, people want a way to say “stop this from ticking on me please”. Which may be totally separate from what I'm asking

**Comment by rcaudy** [comment]
Related, we could improve `snapshot` behavior for immutable columns. See existing ticket: https://github.com/deephaven/deephaven-core/issues/4229

**Comment by rcaudy** [comment]
For non-immutable columns, we could introduce lazy-copying column sources that present a consistent previous view of the source data, writing down the previous values if and when they change. This table would need to behave as a refreshing table for correctness purposes, but would never need to fire an update.

It's not an especially small body of work.
